Feature/web parser

cbrandt6 opened the pull request.

cbrandt6 committed:
Created file for web parser

cbrandt6 committed:
Copied over applicable code from vis parser

cbrandt6 committed:
Added parsed json to gitignore

cbrandt6 committed:
Prelimenary web parser

cbrandt6 committed:
Extracting dolphin id and date from file path

cbrandt6 committed:
Extract time from file

Converted to a function as well

cbrandt6 committed:
Changed regex to better match date

Also changed the join to skip the first character

cbrandt6 committed:
Cleaning up comments

cbrandt6 committed:
Changed robot id regex to exclude 'LOG'

cbrandt6 committed:
Removed connected robots

Mnenmenth commented on src/web_parser.py:
```diff
@@ -0,0 +1,73 @@
+
+    # Parsed script data
+    parsed = {
+        "robot_id": robot_id,
```
It would probably be better for `robot_id` to be called something else here since `Narwhal` could also be the value here

Mnenmenth replied:
This also applies to the `robot_id` variable(s) above when attempting to find if the log is a Dolphin or Narwhal

cbrandt6 replied:
My thinking was the Narwhal is still a robot, just not a Dolphin.

Mnenmenth replied:
The Narwhal is the ground control station that all of the robots communicate with, not really a robot itself

Mnenmenth commented on src/web_parser.py:
```diff
@@ -0,0 +1,73 @@
+        print("This is a Narwhal log")
+        log_type = "Narwhal"
+        robot_id = "Narwhal"
+    elif "LOG_Dolphin" in file_path:
```
Is `file_path` used here on purpose or did you mean to use `file.name`?

cbrandt6 replied:
Nope this was a mistake. Good catch.

Mnenmenth reviewed (changes requested):
There should be some error handling if "Error searching for robot id" and "Log neither Dolphin nor Narwhal" occurs. It would be good to make sure that all of the line splitting, regex searches, etc. are successful. Also see the comments that I made

cbrandt6 committed:
Making Earl's changes

Changed robot id to device id. Also, added try excepts and sys exits for error cases

Mnenmenth commented on src/web_parser.py:
```diff
@@ -0,0 +1,80 @@
+    if "LOG_Narwhal" in file.name:
+        print("This is a Narwhal log")
+        log_type = "Narwhal"
+        device_id = "Narwhal"
```
Much better name 👍

cbrandt6 committed:
Comments

Mnenmenth reviewed (approved):
Good stuff 👍

Walrussuit101 reviewed (approved):
Tested, good stuff :+1:

cbrandt6 committed:
Removed a generated script

cbrandt6 committed:
Merge branch 'develop' into feature/webParser
